### DISCUSSION TITLE
Centering file title

### DISCUSSION THREAD

**Discussion by bryant-the-coder** [nvim-lualine/lualine.nvim#451] [Configuration]
I am trying to center my file title in the middle. how do i achieve that?
CODE:
```
require 'lualine' .setup {
    options = {
      icons_enabled = true,
      theme = 'auto',
      component_separators = { left = '', right = ''},
      section_separators = { left = '', right = ''},
      disabled_filetypes = {'nvim-tree'},
      always_divide_middle = true,
    },

    sections = {
      lualine_a = {'mode'},
      lualine_b = {'branch'},
      lualine_c = { {'filename', file_status = true} },
      lualine_x = {'encoding', 'fileformat', 'filetype'},
      lualine_y = {'progress'},
      lualine_z = {'location'}
    },
    tabline = {},
    extensions = {'nvim-tree','toggleterm'}
  }
```

**Comment by shadmansaleh** [comment]
Try
```diff
-lualine_c = { {'filename', file_status = true} },
+lualine_c = { '%=', {'filename', file_status = true} },
```

**Reply by rubenmate** [reply]
I was looking for this, thanks!

**Comment by bryant-the-coder** [comment]
```
require 'lualine' .setup {
    options = {
      icons_enabled = true,
      theme = 'auto',
      component_separators = { left = '', right = ''},
      section_separators = { left = '', right = ''},
      disabled_filetypes = {'nvim-tree'},
      always_divide_middle = true,
    },

    sections = {
      lualine_a = { 'mode' },
      lualine_b = {'diagnostics', update_in_insert = true, always_visible = true,},
      lualine_c = { '%=', {'filename', file_status = true} },
      lualine_x = {'filetype'},
      lualine_y = {'location'},
      lualine_z = {'progress'},
    },
    tabline = {},
    extensions = {'nvim-tree','toggleterm'}
  }

```
is this correct? for the diagnostics part... no diagnostics showing up

```
### Unrecognized component
Only functions, strings and tables can be used as component.
You seem to have a `boolean` as component indexed as `update_in_insert`.
Something like:
```lua
    update_in_insert = true,
```

This commonly occurs when you forget to pass table with option for component.
When a component has option that component needs to be a table which holds
the component as first element and the options as key value paris.
For example:
```lua
lualine_c = {
    {'diagnostics',
        sources = {'nvim'},
    }
}

lualinenotice
```
Notice the inner extra {} surrounding the component and it's options.
Make sure your config follows this.

### Unrecognized component
Only functions, strings and tables can be used as component.
You seem to have a `boolean` as component indexed as `always_visible`.
Something like:
```lua
    always_visible = true,
```

This commonly occurs when you forget to pass table with option for component.
When a component has option that component needs to be a table which holds
the component as first element and the options as key value paris.
For example:
```lua
lualine_c = {
    {'diagnostics',
        sources = {'nvim'},
    }
}
```
Notice the inner extra {} surrounding the component and it's options.
Make sure your config follows this.
```

**Reply by shadmansaleh** [reply]
The message tells you exactly what's wrong with your config and the fix too . Read it carefully.

**Reply by bryant-the-coder** [reply]
I followed the conf in the readme...

**Reply by shadmansaleh** [reply]
>I followed the conf in the readme...

Which part ?

Readme does show correct way to do it . This is such a common mistake people make in there config that I added a warring directly in lualine still getting issues :P 

```diff
-lualine_b = {'diagnostics', update_in_insert = true, always_visible = true,},
+lualine_b = {{'diagnostics', update_in_insert = true, always_visible = true,}},
```

like the message says a components options must be in components table not in the section table.

**Reply by bryant-the-coder** [reply]
Thanks. sorry for inconvenience. Where can i share my lualine config?
